org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -1278,8 +1281,11 @@ public static Preferences createFrom(Map<String, Object> configuration) {
 		prefs.setNonnullbydefaultTypes(nonullbydefaultTypes);
 		String nullAnalysisMode = getString(configuration, JAVA_COMPILE_NULLANALYSIS_MODE, null);
 		prefs.setNullAnalysisMode(FeatureStatus.fromString(nullAnalysisMode, FeatureStatus.disabled));
-		List<String> cleanupActionsOnSave = getList(configuration, JAVA_CLEANUPS_ACTIONS_ON_SAVE, Collections.emptyList());
+		List<String> cleanupActionsOnSaveTemp = getList(configuration, JAVA_CLEANUPS_ACTIONS_ON_SAVE_DEPRECATED, Collections.emptyList());
+		List<String> cleanupActionsOnSave = getList(configuration, JAVA_CLEANUPS_ACTIONS, cleanupActionsOnSaveTemp);
```

Comment:
`getList(..)` only return the default value specified (in this case `cleanupActionsOnSaveTemp`) if the `configuration` doesn't contain a key `JAVA_CLEANUPS_ACTIONS` whose value is a `String` or `List` type. If you look at https://github.com/redhat-developer/vscode-java/pull/3451/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R1215 you'll see you set the default value to `[]`, which ends up here as an empty list.

So the "deprecated" setting is never respected.

I think if you set both `getList(..)` to have their default be `Collections.emptyList()` then just set `cleanupActionsOnSave` to `cleanupActionsOnSaveTemp` **only** if `cleanupActionsOnSave` is empty  and `cleanupActionsOnSaveTemp` is not empty.

Also when you finish that, rename `cleanupActionsOnSave` to just `cleanupActions`. Same for its getter/setter.